redhat-cop/automate_tower_genie_collections
# Let the collection to use both awx.awx and ansible.controller collections, as per user decision

add variable to set the plugin name to use everywhere

### What does this PR do?
Right now, the tasks in filetree_create and object_diff roles are using the `ansible.controller` collection to make use of the `controller_api` plugin, so anyone trying to use the `awx.awx` collection will get in trouble as the `ansible.controller` collection should not be present. To let the end users to chose freely what collection they want to use, I've modified the code to get that collection name from a variable, which must be defined with the correct value to work successfully.

### How should this be tested?
```
$ cat filetree_create.yml
---
- hosts: localhost
  connection: local
  gather_facts: false
  vars:
    controller_hostname: "{{ vault_controller_hostname }}"
    controller_username: "{{ vault_controller_username }}"
    controller_password: "{{ vault_controller_password }}"
    controller_validate_certs: "{{ vault_controller_validate_certs }}"
    controller_api_plugin: "ansible.controller.controller_api"
  pre_tasks:
    - name: "Check if the required input variables are present"
      assert:
        that:
          - input_tag is defined
          - (input_tag | type_debug) == 'list'
        fail_msg: 'A variable called ''input_tag'' of type ''list'' is needed: -e ''{input_tag: [organizations, projects]}'''
        quiet: true

    - name: "Check if the required input values are correct"
      assert:
        that:
          - tag_item in valid_tags
        fail_msg: "The valid tags are the following ones: {{ valid_tags | join(', ') }}"
        quiet: true
      loop: "{{ input_tag }}"
      loop_control:
        loop_var: tag_item
  roles:
    - redhat_cop.controller_configuration.filetree_create
...
$ ansible-playbook filetree_create.yml -e '{input_tag: [inventory], output_path: ~/iam/filetree_create.2.1.7 }'
```

### Is there a relevant Issue open for this?
No issue opened for this.

### Other Relevant info, PRs, etc.
N/A
